pkg/encoding/fix_test.go

```diff
@@ -0,0 +1,184 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encoding
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/kedgeproject/kedge/pkg/spec"
+
+	api_v1 "k8s.io/client-go/pkg/api/v1"
+)
+
+func TestFixServices(t *testing.T) {
+	failingTest := []spec.ServiceSpecMod{
```

Comment:
Please see the code from Kubernetes again, the tests are separated on passing and failing behavior.

Passing tests, where output is matched: [here](https://github.com/kubernetes/apimachinery/blob/17c1f5d09ee69b7521aa62b2c9d641cdc0dc74e9/pkg/api/resource/quantity_test.go?utf8=%E2%9C%93#L227)

Failing tests, where you will certainly get an error and you are not matching anything just checking if the function being tested errored out: [here](https://github.com/kubernetes/apimachinery/blob/17c1f5d09ee69b7521aa62b2c9d641cdc0dc74e9/pkg/api/resource/quantity_test.go?utf8=%E2%9C%93#L478)